### PR TITLE
Monthly external

### PR DESCRIPTION
**Pull request by aptalca**
- Convert the templating to a 3 stage process:
  - Update Jenkinsfile
  - Delete old templates
  - Update the rest of the templates
- Move external templates (readme lite, documentation and unraid template) to a folder named `.jenkins-external` so they can be easily deleted when run locally
- Add `.jenkins-external` to `.gitignore` if not already in there
- Enable docker-ci updates and fix the ghcr image name in Jenkinsfiles (currently incorrectly pointing to branch pushed images) (requires https://github.com/linuxserver/docker-ci/pull/16 merged first)
- Enable readme-sync updates and fix the ghcr image name in Jenkinsfiles (currently incorrectly pointing to branch pushed images) (requires https://github.com/linuxserver/docker-readme-sync/pull/20 merged first)
- Rebase to alpine 3.13, use the ghcr.io endpoint for baseimage
- Prepend github releases with branch/tag if not latest

### CODE REVIEW

File: Jenkinsfile
Comment (nemchik) [inline Jenkinsfile:268]:
```suggestion
                  git rm "${i}" || :
```
Does this need an "or true" of sorts? https://github.com/linuxserver/docker-jenkins-builder/pull/90/files#diff-e6ffa5dc854b843b3ee3c3c28f8eae2f436c2df2b1ca299cca1fa5982e390cf8L252 seems to have it.

File: Jenkinsfile
Comment (nemchik) [inline Jenkinsfile:242]: Do we always want to use the master branch? Or should we
```suggestion
                git checkout -f "${BRANCH_NAME}"
```
Or would it be `git checkout -f {{ ls_branch }}` ?

File: Jenkinsfile
Comment (nemchik) [inline Jenkinsfile:265]: Do we always want to use the master branch? Or should we
```suggestion
                git checkout -f "${BRANCH_NAME}"
```
Or would it be `git checkout -f {{ ls_branch }}` ?

File: Jenkinsfile
Comment (nemchik) [inline Jenkinsfile:303]:
```suggestion
              if [[ ("${BRANCH_NAME}" == "master") || ("${BRANCH_NAME}" == "main") ]] && [[ (! -f ${TEMPDIR}/gitbook/docker-documentation/images/docker-${CONTAINER_NAME}.md) || ("$(md5sum ${TEMPDIR}/gitbook/docker-documentation/images/docker-${CONTAINER_NAME}.md | awk '{ print $1 }')" != "$(md5sum ${TEMPDIR}/docker-${CONTAINER_NAME}/.jenkins-external/docker-${CONTAINER_NAME}.md | awk '{ print $1 }')") ]]; then
                cp ${TEMPDIR}/docker-${CONTAINER_NAME}/.jenkins-external/docker-${CONTAINER_NAME}.md ${TEMPDIR}/gitbook/docker-documentation/images/
```
I think `.jenkins-external` would be more precise and less likely to conflict with a folder that could exist in the image repos, and easier to add to `.gitignore` files.

File: README.md
Comment (nemchik) [inline README.md:24]:
```suggestion
rm -rf .jenkins-external"
```

File: readme-vars.yml
Comment (nemchik) [inline readme-vars.yml:27]:
```suggestion
  rm -rf .jenkins-external"
```

File: roles/generate-jenkins/tasks/main.yml
Comment (nemchik) [inline roles/generate-jenkins/tasks/main.yml:121]:
```suggestion
    path: "/tmp/.jenkins-external"
```

File: roles/generate-jenkins/tasks/main.yml
Comment (nemchik) [inline roles/generate-jenkins/tasks/main.yml:130]:
```suggestion
    path: "jenkins/{{ project_repo_name }}/.jenkins-external"
```

File: roles/generate-jenkins/templates.yml
Comment (nemchik) [inline roles/generate-jenkins/templates.yml:6]:
```suggestion
 - { src: 'lite.j2', dest: '.jenkins-external/README.lite' , readme: 'true' }
 - { src: 'DOCUMENTATION.j2', dest: ".jenkins-external/{{ project_repo_name }}.md" , readme: 'true' }
```

File: roles/generate-jenkins/templates.yml
Comment (nemchik) [inline roles/generate-jenkins/templates.yml:18]:
```suggestion
 - { src: 'UNRAID_TEMPLATE.j2', dest: ".jenkins-external/{{ project_name | lower }}.xml" , unraid_template: 'true' }
```

File: roles/generate-jenkins/templates/Jenkinsfile.j2
Comment (nemchik) [inline roles/generate-jenkins/templates/Jenkinsfile.j2:467]:
```suggestion
              if [[ ("${BRANCH_NAME}" == "master") || ("${BRANCH_NAME}" == "main") ]] && [[ (! -f ${TEMPDIR}/gitbook/docker-documentation/images/docker-${CONTAINER_NAME}.md) || ("$(md5sum ${TEMPDIR}/gitbook/docker-documentation/images/docker-${CONTAINER_NAME}.md | awk '{ print $1 }')" != "$(md5sum ${TEMPDIR}/docker-${CONTAINER_NAME}/.jenkins-external/docker-${CONTAINER_NAME}.md | awk '{ print $1 }')") ]]; then
                cp ${TEMPDIR}/docker-${CONTAINER_NAME}/.jenkins-external/docker-${CONTAINER_NAME}.md ${TEMPDIR}/gitbook/docker-documentation/images/
```

File: roles/generate-jenkins/templates/Jenkinsfile.j2
Comment (nemchik) [inline roles/generate-jenkins/templates/Jenkinsfile.j2:478]:
```suggestion
                sed -i "s|master/linuxserver.io/img/linuxserver-ls-logo.png|master/linuxserver.io/img/${CONTAINER_NAME}-icon.png|" ${TEMPDIR}/docker-${CONTAINER_NAME}/.jenkins-external/${CONTAINER_NAME}.xml
```

File: roles/generate-jenkins/templates/Jenkinsfile.j2
Comment (nemchik) [inline roles/generate-jenkins/templates/Jenkinsfile.j2:480]:
```suggestion
              if [[ ("${BRANCH_NAME}" == "master") || ("${BRANCH_NAME}" == "main") ]] && [[ (! -f ${TEMPDIR}/unraid/templates/unraid/${CONTAINER_NAME}.xml) || ("$(md5sum ${TEMPDIR}/unraid/templates/unraid/${CONTAINER_NAME}.xml | awk '{ print $1 }')" != "$(md5sum ${TEMPDIR}/docker-${CONTAINER_NAME}/.jenkins-external/${CONTAINER_NAME}.xml | awk '{ print $1 }')") ]]; then
```

File: roles/generate-jenkins/templates/Jenkinsfile.j2
Comment (nemchik) [inline roles/generate-jenkins/templates/Jenkinsfile.j2:484]:
```suggestion
                  cp ${TEMPDIR}/docker-${CONTAINER_NAME}/.jenkins-external/${CONTAINER_NAME}.xml ${TEMPDIR}/unraid/templates/unraid/
```

File: roles/generate-jenkins/templates/Jenkinsfile.j2
Comment (nemchik) [inline roles/generate-jenkins/templates/Jenkinsfile.j2:432]:
```suggestion
                  git rm "${i}" || :
```

Does this need an "or true" of sorts? https://github.com/linuxserver/docker-jenkins-builder/pull/90/files#diff-e6ffa5dc854b843b3ee3c3c28f8eae2f436c2df2b1ca299cca1fa5982e390cf8L252 seems to have it.

File: Jenkinsfile
Comment (aptalca) [reply Jenkinsfile:268]: shouldn't be needed because we check for its existence right above when we set the var `TEMPLATES_TO_DELETE`

File: Jenkinsfile
Comment (aptalca) [reply Jenkinsfile:303]: that's a good point

File: Jenkinsfile
Comment (aptalca) [reply Jenkinsfile:242]: it is, but this is the Jenkinsfile, not the Jenkinsfile.J2 template 😉

File: roles/generate-jenkins/templates/Jenkinsfile.j2
Comment (aptalca) [reply roles/generate-jenkins/templates/Jenkinsfile.j2:432]: Not needed because the file's existence is checked right above, before the var is set

File: Jenkinsfile
Comment (aptalca) [reply Jenkinsfile:265]: this is the jenkinsfile, not the template

File: Jenkinsfile
Comment (aptalca) [reply Jenkinsfile:303]: fixed in the template

File: README.md
Comment (aptalca) [reply README.md:24]: will fix in the template

**Comment by nemchik** [comment]
Looks good. Can we put the gitignore bit in there too?

With that I'll approve once I see the other two PRs mentioned merged.

**Review by nemchik — APPROVED** [review]

**Pull request merged**
